class-auth.php

```diff
@@ -321,11 +321,15 @@ public function validate_token( $return_response = true ) {
 			$auth = isset( $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] ) ? $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] : false;
 		}
 
-		$hash = end(explode("_", AUTH_COOKIE));
+        $auth_arr = explode("_", AUTH_COOKIE);
+
+		$hash = end($auth_arr);
 
+		//Check if basic auth is also present
 		if(! $auth) {
 			foreach(array_keys($_COOKIE) as $key) {
```

Comment:
Is it possible to access the auth token using key index access instead of a loop?

```
$token  = $_COOKIE[$hash]; // not sure what $hash is assuming it's the cookie name for the token?

if($token) ...
```

Reply:
I attempted that when we first setup the plugin, I think I was running into issues involving something inconsistent while we were trying to automate grabbing a jwt token for our endpoint check

I can create a ticket and dig into it further

Reply:
Main thing is that this code may run every request so even with a limit being on cookie size it is an unnecessary loop.